core/primitives/tipset/tipset.cpp

```diff
@@ -59,7 +59,31 @@ namespace fc::primitives::tipset {
                                               const Visitor &visitor) {
     auto onMessage = [&](auto bls, auto &cid) -> outcome::result<void> {
       if (visited.insert(cid).second) {
-        OUTCOME_TRY(visitor(visited.size() - 1, bls, cid));
+        SignedMessage smsg;
+        auto &msg{smsg.message};
+        if (load) {
+          if (bls) {
+            OUTCOME_TRYA(msg, ipld->getCbor<UnsignedMessage>(cid));
+          } else {
+            OUTCOME_TRYA(smsg, ipld->getCbor<SignedMessage>(cid));
```

Comment:
Why UnsignedMessage if bls is true?
I thought that bls means signature

Reply:
`SignedMessage` = `UnsignedMessage` + `Signature`.
Block = secp `SignedMessage` + bls `UnsignedMessage` + aggregated bls `Signature`

Reply:
Ok